WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/VideoPressBlockProcessor.kt

```diff
@@ -0,0 +1,198 @@
+            queryArgs["loop"] = "true"
+        }
+    }
+
+    // Adds Volume-related options. Muted: Turned OFF by default.
+    private fun addMutedAndPersistVolumeArgs(queryArgs: MutableMap<String, String>, muted: Boolean?) {
+        if (muted == true) {
+            queryArgs["muted"] = "true"
+            queryArgs["persistVolume"] = "false"
+        }
+    }
+
+    // Adds PlaysInline option. Turned OFF by default.
+    private fun addPlaysinlineArg(queryArgs: MutableMap<String, String>, playsinline: Boolean?) {
+        if (playsinline == true) {
+            queryArgs["playsinline"] = "true"
+        }
+    }
+
+    // Adds Poster option. No image by default.
+    private fun addPosterArg(queryArgs: MutableMap<String, String>, poster: String?) {
+        poster?.let { queryArgs["posterUrl"] = it }
+    }
+
+    // Adds Preload option. Metadata by default.
+    private fun addPreloadArg(queryArgs: MutableMap<String, String>, preload: String?) {
+        preload?.let { if (it.isNotEmpty()) queryArgs["preloadContent"] = it }
+    }
+
+    /**
+     * Adds Seekbar options.
+     * - SeekbarColor: No color by default.
+     * - SeekbarPlayerColor: No color by default.
+     * - SeekbarLoadingColor: No color by default.
+     */
+    private fun addSeekbarArgs(
+        queryArgs: MutableMap<String, String>,
+        seekbarColor: String?,
+        seekbarPlayedColor: String?,
+        seekbarLoadingColor: String?
+    ) {
+        seekbarColor?.let { if (it.isNotEmpty()) queryArgs["sbc"] = it }
+        seekbarPlayedColor?.let { if (it.isNotEmpty()) queryArgs["sbpc"] = it }
+        seekbarLoadingColor?.let { if (it.isNotEmpty()) queryArgs["sblc"] = it }
+    }
+
+    // Adds UseAverageColor option. Turned ON by default.
+    private fun addUseAverageColorArg(queryArgs: MutableMap<String, String>, useAverageColor: Boolean?) {
+        if (useAverageColor == true) {
+            queryArgs["useAverageColor"] = "true"
+        }
+    }
+
+    override fun processBlockContentDocument(document: Document?): Boolean {
+        val videoPressElements = document?.select("figure")
+
+        if (videoPressElements != null) {
+            val url = getVideoPressURL(mRemoteGuid)
+            videoPressElements.append("<div class='jetpack-videopress-player__wrapper'>$url</div>")
+            return true
+        }
+
+        return false
+    }
+
+    override fun processBlockJsonAttributes(jsonAttributes: JsonObject?): Boolean {
+        val id = jsonAttributes?.get(ID_ATTRIBUTE)
+        val src = jsonAttributes?.get(SRC_ATTRIBUTE)
+
+        return if (id != null && !id.isJsonNull && id.asString == mLocalId) {
+            jsonAttributes.apply {
+                addProperty(ID_ATTRIBUTE, Integer.parseInt(mRemoteId))
+                addProperty(GUID_ATTRIBUTE, mRemoteGuid)
+                when (src) {
+                    null -> Unit
+                    else -> jsonAttributes.addProperty(SRC_ATTRIBUTE, mRemoteUrl)
+                }
```

Comment:
In the [iOS implementation](https://github.com/wordpress-mobile/WordPress-iOS/blob/0e5eac244218a31a90e23fb9ef8fa82967c2bb48/WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergVideoPressUploadProcessor.swift#L65-L84) we aren't processing the `src` attribute, so not sure if we need to assign the remote URL here 🤔 .

Reply:
In my testing the `src` attribute was being added on Android when I replaced the video. From the results of your testing, it seems there are some wider issues with replacing videos, so I'll do some more digging and reply here when I've figured out what's happening. 🤔 Thanks for your thorough review!

Reply:
Right, the `src` attribute is being added with the local file URL. This is needed to determine that the upload is in progress. However, I'm not sure if we need to update the value in the processor, I think we can leave it with its original value, at least that's what I did in the iOS implementation. Alternatively, we could remove it when the upload finishes, as [we do in the block logic](https://github.com/Automattic/jetpack/blob/3b4d09a3bc4c4aabd08d532d214e22c0bd7bf17d/projects/packages/videopress/src/client/block-editor/blocks/video/edit.native.tsx#L136).

Reply:
Ah! I see, thank you for expanding. I agree it'd be best to match the iOS implementation, so I've removed this code in https://github.com/wordpress-mobile/WordPress-Android/pull/18187/commits/f8cd6a8865370c1c131382a23c49135d4a95682b. 🙇‍♀️